src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,381 @@
+     * @dev `who` address should have been whitelisted using `kiss`.
+     */
+    function pick(address who) external {
+        require(can[msg.sender] == 1, "RwaOutputConduit3/not-operator");
+        require(bud[who] == 1 || who == address(0), "RwaOutputConduit3/not-bud");
+        to = who;
+        emit Pick(who);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Method to swap DAI contract balance to GEM through PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     */
+    function push() external isMate {
+        _doPush(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     * @param wad DAI amount.
+     */
+    function push(uint256 wad) external isMate {
+        _doPush(wad);
+    }
+
+    /**
+     * @notice Flushes out any DAI balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external isMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `usr` address.
+     * @param usr Destination address
+     * @dev Can be called only by admin.
+     */
+    function yank(address usr) external auth {
+        uint256 amt = gem.balanceOf(address(this));
+        gem.transfer(usr, amt);
+        emit Yank(usr, amt);
+    }
+
+    /**
+     * @notice Calculate expected amount of GEM received for swapping `wad` amount of DAI.
+     * @param wad DAI amount.
+     * @return gemAmt Amount of GEM expected.
+     */
+    function expectedGemAmt(uint256 wad) public view returns (uint256) {
+        return mul(wad, WAD) / mul(add(WAD, psm.tout()), to18ConvertionFactor);
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @param wad DAI amount.
+     */
+    function _doPush(uint256 wad) internal {
+        require(to != address(0), "RwaOutputConduit3/to-not-picked");
+
+        // We might lose some dust here because of rounding errors. I.e.: USDC has 6 dec and DAI has 18.
+        uint256 gemAmt = expectedGemAmt(wad);
+        require(gemAmt > 0, "RwaOutputConduit3/insufficient-swap-gem-amount");
+
+        psm.buyGem(address(this), gemAmt);
+
+        address recipient = to;
+        to = address(0);
+
+        gem.transfer(recipient, gemAmt);
```

Comment:
```suggestion
        address recipient = to;
        to = address(0);

        psm.buyGem(recipient, gemAmt);
```

Reply:
done

Reply:
Hmm change pushed wasn't what I'm suggesting.

Reply:
In my solution there isn't `gem.transfer` you just buy directly to the recipient avoiding store operations.

Reply:
I see... Fixed

Reply:
Looks good now.